kubernetes/config-maps/nginx-storage/default.conf.j2

```diff
@@ -207,14 +208,34 @@ server {
 
             os.rename("/srv/storage/" .. tmp_file_name, "/srv/storage/" .. header_path_with_filename)
             os.remove("/srv/storage/" .. tmp_file_name .. ".shactx")
+
+            -- Send confirmation to the conductor after uploaded file.
+            local http = require "resty.http"
+            local httpc = http.new()
+            local res, err = httpc:request_uri("{{ upload_report_url }}" .. header_path_with_filename, {
```

Comment:
Please add a comment somewhere near the definition of `upload_report_url` that it must contain the path. E.g. `http://conductor.default.svc.cluster.local/`, not `http://conductor.default.svc.cluster.local`.

Otherwise someone could give us a filename like `.my.own.domain.com/my-malicious-endpoint` so that we end up making a request to `http://conductor.default.svc.cluster.local.my.own.domain.com/my-malicious-endpoint/` which is outside the cluster!

It would be best if we could have some sanity check for `upload_report_url` so that's impossible to make such a mistake. But at least add a comment so that someone setting it is aware of it.

Reply:
I added a comment in `concent-deployment-values` repositories in `var.yml` file.

Reply:
``` patch
-# Address of conductor endpoint for sending notification after successfully upload used by nginx-storage
+# Address of conductor endpoint for sending notification when a file is uploaded used by nginx-storage
+# The string with url path must end with slash, for example "http://conductor.default.svc.cluster.local/conductor/report-upload/",
+# not "http://conductor.default.svc.cluster.local/conductor/report-upload"
 upload_report_url: "http://conductor.default.svc.cluster.local/conductor/report-upload/"
```

That's not entirely true. It does not have to end with a slash. It's enough if it ends with a path (which starts with a slash).

Reply:
How should it work without a slash? This variable is combined with a path to a file that looks like this `blender/result/ef0dc1/ef0dc1.zzz523.zip`, so when combinate it together you receive: `http://conductor.default.svc.cluster.local/conductor/report-uploadblender/result/ef0dc1/ef0dc1.zzz523.zip`

Reply:
OK. You're right. It must end with a slash. I did not notice it before. If there's a path but no slash the attacker won't be able to change the domain but could still make nginx-storage connect to a different path. The risk that he'll be able to do anything malicious that way is much lower than when changing a domain name but still it's a valid reason to require a slash.

But please explain both cases in a comment.